Lib/zipfile/_path.py

```diff
@@ -281,9 +281,9 @@ def stem(self):
     def filename(self):
         return pathlib.Path(self.root.filename).joinpath(self.at)
 
-    def read_text(self, *args, **kwargs):
-        kwargs["encoding"] = io.text_encoding(kwargs.get("encoding"))
-        with self.open('r', *args, **kwargs) as strm:
+    def read_text(self, encoding=None, *args, **kwargs):
+        encoding = io.text_encoding(encoding)
```

Comment:
I try to avoid mutating variables, though I guess mutating `locals()['encoding']` is not much worse than mutating `kwargs['encoding']`.

Reply:
if we keep any logic for this in read_text at all (given that it calls open which also handles this case), i'll just inline the io.text_encoding call within the parameter list...